Admin/Action.php

```diff
@@ -1,70 +1,89 @@
 <?php
 
-namespace BlueBear\AdminBundle\Admin;
+namespace LAG\AdminBundle\Admin;
 
-class Action 
+use LAG\AdminBundle\Admin\Configuration\ActionConfiguration;
+
+class Action implements ActionInterface
 {
     /**
+     * Action name.
+     *
      * @var string
      */
     protected $name;
 
     /**
+     * Action title.
+     *
      * @var string
      */
     protected $title;
 
     /**
+     * Fields displayed for this action.
+     *
      * @var Field[]
      */
     protected $fields = [];
 
     /**
+     * Action permissions.
+     *
      * @var string[]
      */
     protected $permissions = [];
 
-    protected $route;
-
-    protected $parameters;
+    /**
+     * Configured linked actions to display in this view.
+     *
+     * @var array
+     */
+    protected $actions = [];
 
     /**
-     * Export types
+     * Actions displayed at the bottom of the view.
      *
      * @var array
      */
-    protected $export = [];
+    protected $submitActions = [];
+
+    protected $configuration;
 
     /**
-     * @return string
+     * @var array
      */
-    public function getName()
-    {
-        return $this->name;
-    }
+    protected $filters = [];
 
     /**
-     * @param string $name
+     * @var string[]
      */
-    public function setName($name)
+    protected $batchActions = [];
+
+    public function __construct($actionName, array $actionOptions, ActionConfiguration $configuration)
```

Comment:
Options are already resolved in the ActionFactory at this step of Action creation.